docs/docs/jsonnet/injecting-values.md

```diff
@@ -0,0 +1,89 @@
+---
+name: "Injecting Values"
+route: "/jsonnet/injecting-values"
+menu: "Writing Jsonnet"
+---
+
+# Injecting Values
+
+Sometimes it might be required to pass externally acquired data into Jsonnet.
+
+There are three ways of doing so:
+
+1. [JSON files](#json-files)
+2. [External variables](#external-variables)
+3. [Top level arguments](#top-level-arguments)
+
+## JSON files
+
+Jsonnet is a superset of JSON, it treats any JSON as valid Jsonnet. Because many
+systems can be told to output their data in JSON format, this provides a pretty
+good interface between those.
+
+For example, your deployment pipeline could acquire secrets from systems such as
+[Vault](https://www.vaultproject.io/), etc. and write that into `secrets.json`.
+
+```jsonnet
+local secrets = import "secrets.json";
+
+{
+  foo: secrets.myPassword,
+}
+```
+
+## External variables
+
+Another way of passing values from the outside are external variables, which are specified like so:
+
+```bash
+# strings
+$ tk show . --ext-str hello=world
+
+# any Jsonnet snippet
+$ tk show . --ext-str foo=4 --ext-str bar='[ 1, 3 ]'
+```
+
+They can be accessed using `std.extVar` and the name given to them on the command line:
+
+```jsonnet
+{
+  foo: std.extVar('foo'), // 4, integer
+  bar: std.extVar('bar'), // [ 1, 3 ], array
+}
+```
+
+> **Warning**: It's not possible to detect what extVars are available, and it's
+> not possible to set default values.  
+> Try to use [**Top Level Arguments**](#top-level-arguments) instead, to avoid surprising behaviour
```

Comment:
I would explain why surprising behavior may occur. The way it's currently phrased sounds like the support in Jsonnet or Tanka is not reliable.

Maybe something along these lines:
"""
External variables are directly accessible in all parts of the configuration. It is sometimes difficult to keep track where they're used and what effect they have on the final result. Try to use [**Top Level Arguments**](#top-level-arguments) instead.
"""